### PR TITLE
SAK-41763: XSS in chat user name - 12.x

### PR DESCRIPTION
**Pull request by hornersa**
For this fix to be applied to 12.x, I needed to add some runtime libraries and upgrade taglibs from 1.0.4. I chose 1.1.2 because this is the version already used by the site/admin-prefs tool in 12.x.

### CODE REVIEW

File: chat/chat-tool/tool/pom.xml [hunk 1]
```diff
@@ -98,7 +98,13 @@
...
       <groupId>taglibs</groupId>
       <artifactId>standard</artifactId>
-      <version>1.0.4</version>
+      <version>1.1.2</version>
+      <scope>runtime</scope>
+    </dependency>
+    <dependency>
+        <groupId>javax.servlet</groupId>
+        <artifactId>jstl</artifactId>
+	<scope>runtime</scope>
```
Comment (ern) [inline chat/chat-tool/tool/pom.xml]: the scope looks wrong here i would remove it...

Comment (hornersa) [reply chat/chat-tool/tool/pom.xml]: It looks like the jstl jar isn't needed after all. (That's what I started with when troubleshooting 12.x to compare with trunk where trunk had a jstl jar deployed.) The runtime scope for the taglibs jar doesn't look needed either. I'll revise my commit.

**Comment by mpellicer** [comment]
Thank you so much @hornersa

File: chat/chat-tool/tool/src/webapp/jsp/roomMonitor.jspf [hunk 2]
```diff
@@ -42,7 +42,7 @@
             <c:out value="<span class='chatUserOnline'></span>" escapeXml="false" />
             <c:out value="</span>" escapeXml="false" />
 
-            <c:out value="<span class='chatNameDate'><span class='chatName'>${message.owner}</span>" escapeXml="false" />
+            <c:out value="<span class='chatNameDate'><span class='chatName'>${fn:escapeXml(message.owner)}</span>" escapeXml="false" />
```
Comment (Sjord) [inline chat/chat-tool/tool/src/webapp/jsp/roomMonitor.jspf]: I am not familiar with this templating language, but it seems to me this should be written as follows:

    <span class='chatNameDate'><span class='chatName'><c:out value="${message.owner}"></span>

**Pull request merged**